lib/Sabberworm/CSS/Parser.php

```diff
@@ -64,20 +64,75 @@ public function __construct($sText, Settings $oParserSettings = null, $iLineNo =
 			$this->aSizeUnits[$iSize][strtolower($val)] = $val;
 		}
 		ksort($this->aSizeUnits, SORT_NUMERIC);
+		$this->fixCharset();
 	}
 
-	public function setCharset($sCharset) {
-		$this->sCharset = $sCharset;
-		$this->aText = $this->strsplit($this->sText);
-		$this->iLength = count($this->aText);
+	private function fixCharset() {
+		// We need to know the charset before the parsing starts.
+		$sCharset = strtolower($this->oParserSettings->sDefaultCharset);
+		if (strpos($this->sText, "\xef\xbb\xbf") === 0) {
```

Comment:
Also remove the UTF-16 and -32 BOMs (0xFFFE and 0xFEFF). Throw an exception if the `@charset` declaration does not match the byte order from the DOM.